### PR TITLE
[SYSTEMML-1185] SystemML Breast Cancer Project

### PR DESCRIPTION
**Pull request by dusenberrymw**
This is the initial commit of the SystemML breast cancer project!

Please reference the attached `README.md` for an overview, background information, goals, our approach, etc.

At a high level, this PR introduces the following new files/folders:
* `README.md`: Project information, etc.
* `Preprocessing.ipynb`: PySpark notebook for preprocessing our histopathology slides into an appropriate `DataFrame` for consumption by SystemML.
* `MachineLearning.ipynb`: PySpark/SystemML notebook for our machine learning approach thus far.  We started simple, and are currently in need of engine improvements in order to proceed forward.
* `softmax_clf.dml`: Basic softmax model (multiclass logistic regression with normalized probabilities) as a sanity check.
* `convnet.dml`: Our current deep convnet model.  We are starting simple with a slightly extended "LeNet"-like network architecture.  The goal will be to improve engine performance so that this model can be efficiently trained, and then move on to larger, more recent types of model architectures.
* `hyperparam_tuning.dml`: A separate script for performing a hyperparameter search for our current convnet model.  This has been extracted from the notebook as the current `parfor` engine implementation is not yet sufficient for this type of necessary job.
* `data`: A placeholder folder into which the data could be downloaded.
* `nn`: A softlink that will point to the SystemML-NN library.
* `approach.svg`: Image of our overall pipeline used in `README.md`.

Overall, I want this project to serve as a large-scale, end-to-end machine learning project that can drive necessary core improvements for SystemML.

### CODE REVIEW

**Comment by dusenberrymw** [comment]
cc @frreiss, @bertholdreinwald, @niketanpansare, @deroneriksson, @MadisonJMyers

**Comment by MadisonJMyers** [comment]
LGTM. Excited to bring the SystemML breast cancer project to the community!

File: projects/breast_cancer/README.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+References:  
+[1] https://web.stanford.edu/group/rubinlab/pubs/2243353.pdf  
+[2] See [`Preprocessing.ipynb`](Preprocessing.ipynb).  
+[3] See [`MachineLearning.ipynb`](MachineLearning.ipynb), [`softmax_clf.dml`](softmax_clf.dml), and [`convnet.dml`](convnet.dml).  
+
+![Approach](https://dusenberrymw.github.io/incubator-systemml/img/projects/breast_cancer/approach.svg)
```
Comment (deroneriksson) [inline projects/breast_cancer/README.md]: @dusenberrymw Could you update the image link to point to the main project rather than the fork?

Comment (deroneriksson) [inline projects/breast_cancer/README.md]: This work is really great @dusenberrymw! LGTM!

Comment (dusenberrymw) [inline projects/breast_cancer/README.md]: Yeah, I'll update that when I merge.  Wanted to make sure the image showed up in the PR still as it requires that it be hosted on the gh-pages branch in order to reference it from the README.

**Comment by deroneriksson** [comment]
This work is really great @dusenberrymw! LGTM!
